services/smtp.go

```diff
@@ -30,9 +30,97 @@
  */
 package services
 
-/*
-// SMTP is a placeholder
-func (d *lowDirector) SMTP() func(net.Conn) {
-	return d.Echo()
+import (
+	"crypto/tls"
+	"net"
+
+	"github.com/honeytrap/honeytrap/event"
+	"github.com/honeytrap/honeytrap/pushers"
+)
+
+var (
+	_           = Register("smtp", SMTP)
+	receiveChan = make(chan Message)
+	serverCert  = "/etc/ssl/certs/mailserver.pem"
+	serverKey   = "/etc/ssl/private/mailserver.pem"
+)
+
+// SMTP
+func SMTP(options ...ServicerFunc) Servicer {
+
+	s := &SMTPService{}
+	s.setServer()
```

Comment:
Could you move the setServer code to above SMTPService, and init SMTPService{ srv: server}